File: Sharebook/Server/src/Sharebook.Common/Client/Validators/HelpClientParametersValidator.cs
//namespace WorldFeed.Common.Client.Validators
//{
//    using WorldFeed.Common.Public.Parameters.HelpClient;

//    public interface IHelpClientParametersValidator
//    {
//        void Validate(IGetRateLimitsParameters parameters);

//        void Validate(IGetTwitterConfigurationParameters parameters);

//        void Validate(IGetSupportedLanguagesParameters parameters);

//        void Validate(IGetPlaceParameters parameters);

//        void Validate(IGeoSearchParameters parameters);

//        void Validate(IGeoSearchReverseParameters parameters);
//    }

//    public class HelpClientParametersValidator : IHelpClientParametersValidator
//    {
//        private readonly IHelpClientRequiredParametersValidator helpClientRequiredParametersValidator;

//        public HelpClientParametersValidator(IHelpClientRequiredParametersValidator helpClientRequiredParametersValidator)
//        {
//            this.helpClientRequiredParametersValidator = helpClientRequiredParametersValidator;
//        }

//        public void Validate(IGetRateLimitsParameters parameters)
//        {
//            this.helpClientRequiredParametersValidator.Validate(parameters);
//        }

//        public void Validate(IGetTwitterConfigurationParameters parameters)
//        {
//            this.helpClientRequiredParametersValidator.Validate(parameters);
//        }

//        public void Validate(IGetSupportedLanguagesParameters parameters)
//        {
//            this.helpClientRequiredParametersValidator.Validate(parameters);
//        }

//        public void Validate(IGetPlaceParameters parameters)
//        {
//            this.helpClientRequiredParametersValidator.Validate(parameters);
//        }

//        public void Validate(IGeoSearchParameters parameters)
//        {
//            this.helpClientRequiredParametersValidator.Validate(parameters);
//        }

//        public void Validate(IGeoSearchReverseParameters parameters)
//        {
//            this.helpClientRequiredParametersValidator.Validate(parameters);
//        }
//    }
//}

File: Sharebook/Client/src/app/core/Core/Client/Validators/HelpClientParametersValidator.ts
import {Inject, Injectable, InjectionToken} from "@angular/core";

import {IGetRateLimitsParameters} from "../../../Public/Parameters/HelpClient/GetRateLimitsParameters";
import {IGetTwitterConfigurationParameters} from "../../../Public/Parameters/HelpClient/GetTwitterConfigurationParameters";
import {IGetSupportedLanguagesParameters} from "../../../Public/Parameters/HelpClient/GetSupportedLanguagesParameters";
import {IGetPlaceParameters} from "../../../Public/Parameters/HelpClient/GetPlaceParameters";
import {IGeoSearchParameters} from "../../../Public/Parameters/HelpClient/GeoSearchParameters";
import {IGeoSearchReverseParameters} from "../../../Public/Parameters/HelpClient/GeoSearchReverseParameters";
import {
  HelpClientRequiredParametersValidator,
  IHelpClientRequiredParametersValidator,
  IHelpClientRequiredParametersValidatorToken
} from "./HelpClientRequiredParametersValidator";
import {HelpParameters} from "./parameters-types";

export interface IHelpClientParametersValidator {
  validate(parameters: IGetRateLimitsParameters): void;

  validate(parameters: IGetTwitterConfigurationParameters): void;

  validate(parameters: IGetSupportedLanguagesParameters): void;

  validate(parameters: IGetPlaceParameters): void;

  validate(parameters: IGeoSearchParameters): void;

  validate(parameters: IGeoSearchReverseParameters): void;
}

export const IHelpClientParametersValidatorToken = new InjectionToken<IHelpClientParametersValidator>('IHelpClientParametersValidator', {
  providedIn: 'root',
  factory: () => new HelpClientParametersValidator(Inject(HelpClientRequiredParametersValidator)),
});

@Injectable()
export class HelpClientParametersValidator implements IHelpClientParametersValidator {
  private readonly _helpClientRequiredParametersValidator: IHelpClientRequiredParametersValidator;

  constructor(@Inject(IHelpClientRequiredParametersValidatorToken) helpClientRequiredParametersValidator: IHelpClientRequiredParametersValidator) {
    this._helpClientRequiredParametersValidator = helpClientRequiredParametersValidator;
  }

  public validate(parameters: HelpParameters): void {
    this._helpClientRequiredParametersValidator.validate(parameters);
  }
}
